app/use_cases/create_anonymous_user.rb

```diff
@@ -29,38 +30,58 @@ def create_or_find_auth_user(auth0_id_string, access_level)
     )
   end
 
-  def create_or_find_person(auth_user)
+  def create_or_find_person(auth_user, email_address)
     return auth_user if auth_user.person.present?
 
-    auth_user.person = Person.create!(first_name: auth_user.auth0_id_string,
-                                      last_name: auth_user.auth0_id_string,
-                                      gender: nil,
-                                      mobile_phone: nil,
-                                      role: find_role,
-                                      auth_user: auth_user)
-    auth_user.save!
+    if email_address.present?
+      person = Person.find_by(email: email_address)
+      if person.present? && person.auth_user.blank?
+        ActiveRecord::Base.transaction do
+          person.update!(auth_user: auth_user, email: nil, role: find_role)
+          auth_user.person = person
+          auth_user.save!
+        end
+        return auth_user
+      end
+    end
+
+    create_new_person(auth_user)
+  end
+
+  def create_new_person(auth_user)
+    ActiveRecord::Base.transaction do
+      auth_user.person = Person.create!(first_name: auth_user.auth0_id_string,
```

Comment:
might be saying something stupid here, but won't `auth_user.person.create!` work? in that case you also don't need the transaction (I cannot test it here on my windows machine..)

Reply:
No that doesn't work because `auth_user.person` is `nil`. `auth_user.create_person` should work, however. I'll change it to that.

![chrome_2019-12-03_16-04-15](https://user-images.githubusercontent.com/607965/70062571-92561d80-15e6-11ea-825c-ce264ef71df5.png)